src/components/Accordion.js

```diff
@@ -0,0 +1,93 @@
+const menuOpen = "collapse show";
+const buttonOpenAll = "Open All";
+const buttonCloseAll = "Close All";
+
+if (!String.prototype.includes) {
+  String.prototype.includes = function(search, start) {
+    "use strict";
+    if (typeof start !== "number") {
+      start = 0;
+    }
+
+    if (start + search.length > this.length) {
+      return false;
+    } else {
+      return this.indexOf(search, start) !== -1;
+    }
+  };
+}
+
+document.addEventListener(
+  "click",
+  clickEvent => {
+    const { target } = clickEvent;
+
+    if (target && target.className === "dg_allitems") {
+      AllMenuItemsAction(target);
+    } else if (target) {
+      MenuAction(target);
+    } else {
+      return;
+    }
+  },
+  false
+);
+
+function MenuAction(element) {
+  var mainDiv = element.parentElement.parentElement;
+  var menuItems = mainDiv.getElementsByClassName("multi-collapse");
+  var childDiv = element.nextElementSibling;
+  var menuID = childDiv.id;
+
+  for (i = 0; i < menuItems.length; i++) {
+    if (menuID !== menuItems[i].id) {
+      menuItems[i].className = mainDiv.className.includes(
+        "dg_allowmutlipleopen"
+      )
+        ? menuItems[i].className
+        : menuState("close");
+    } else {
+      if (menuItems[i].className.includes(menuOpen)) {
+        menuItems[i].className = menuState("close");
+        menuItems[i].setAttribute("aria-expanded", false);
+        element.parentElement.className = "card fa collapsed dg_menuitem";
+      } else {
+        menuItems[i].className = menuState("open");
+        menuItems[i].setAttribute("aria-expanded", true);
+        element.parentElement.className = "card fa dg_menuitem";
+      }
+    }
+  }
+}
+
+function AllMenuItemsAction(button) {
+  var status = button.innerHTML.trim();
+  var body = button.parentElement;
+  var menuItems = body.getElementsByClassName("multi-collapse");
+
+  if (button.innerHTML.trim() === buttonOpenAll) {
+    button.innerHTML = buttonCloseAll;
+  } else {
+    button.innerHTML = buttonOpenAll;
+  }
+
+  for (i = 0; i < menuItems.length; i++) {
+    if (status == "Open All") {
+      menuItems[i].className = menuState("open");
+      menuItems[i].setAttribute("aria-expanded", true);
+      menuItems[i].parentElement.className = "card fa dg_menuitem";
+    } else {
+      menuItems[i].className = menuState("close");
+      menuItems[i].setAttribute("aria-expanded", false);
+      menuItems[i].parentElement.className = "card fa collapsed dg_menuitem";
+    }
+  }
+}
+
+function menuState(state) {
+  if (state === "open") {
```

Comment:
this can be simplified to use a tenary and be returned like this.

```js
const menuState = (state) => state.toLowerCase() === "open" ? "multi-collapse collapse show" : "multi-collapse collapse";
```

this can also probably be refactored since you have "multi-collapse collapse", maybe even just putting that in the html?

Reply:
mutli-collapse and collapse is what we use to open multiple accordian elements. Mutli-collpase is how we gather all the elements and collpase with show or nothing opens closes div. Not sure I understand what you are asking for with a re-factor here.